File: firefly-example/src/main/java/com/firefly/example/reactive/coffee/store/service/impl/UserServiceImpl.java
package com.firefly.example.reactive.coffee.store.service.impl;

import com.firefly.$;
import com.firefly.annotation.Component;
import com.firefly.annotation.Inject;
import com.firefly.example.reactive.coffee.store.dao.UserDAO;
import com.firefly.example.reactive.coffee.store.model.User;
import com.firefly.example.reactive.coffee.store.service.UserService;
import reactor.core.publisher.Mono;

/**
 * @author Pengtao Qiu
 */
@Component
public class UserServiceImpl implements UserService {

    @Inject
    private UserDAO userDAO;

    @Override
    public Mono<User> getByName(String name) {
        return userDAO.getByName(name);
    }
}

File: firefly-example/src/main/kotlin/com/firefly/example/kotlin/coffee/store/service/impl/UserServiceImpl.kt
package com.firefly.example.kotlin.coffee.store.service.impl

import com.firefly.annotation.Component
import com.firefly.annotation.Inject
import com.firefly.example.kotlin.coffee.store.dao.UserDAO
import com.firefly.example.kotlin.coffee.store.model.User
import com.firefly.example.kotlin.coffee.store.service.UserService

/**
 * @author Pengtao Qiu
 */
@Component
class UserServiceImpl : UserService {

    @Inject
    lateinit var userDAO: UserDAO

    suspend override fun getByName(name: String): User = userDAO.getByName(name)

}